### PR TITLE
708 revisit timerange trackers

### PR DESCRIPTION
**Pull request by StaehliJ**
# Pull request

## Description

The goal of this PR is to split blocked time range tracker from credits and chapter time ranges as they can be retrieve in a different way.

## Changes made
- No "public" changes.

## Checklist

- [x] Your branch has been rebased onto the `main` branch.
- [ ] APIs have been properly documented (if relevant).
- [ ] The documentation has been updated (if relevant).
- [ ] New unit tests have been written (if relevant).
- [ ] The demo has been updated (if relevant).
- [ ] All pull request status checks pass.

### CODE REVIEW

File: pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/PillarboxMediaMetaDataTracker.kt [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                    send()
+                }
+                playerMessages.add(messageEnter)
+                playerMessages.add(messageExit)
+            }
+            return playerMessages
+        }
+
+        fun clear() {
+            messages.forEach { it.cancel() }
```
Comment (MGaetan89) [inline pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/PillarboxMediaMetaDataTracker.kt:135]: Shouldn't we clear the list of messages too?

Comment (StaehliJ) [reply pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/PillarboxMediaMetaDataTracker.kt:135]: We can't, messages is a immutable list and they are recreated when needed.

Comment (MGaetan89) [reply pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/PillarboxMediaMetaDataTracker.kt:135]: 🤦🏻‍♂️ silly me!

File: pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/PillarboxExoPlayer.kt [hunk 2]
```diff
@@ -356,11 +338,32 @@ class PillarboxExoPlayer internal constructor(
...
+                listener.onChapterChanged(timeRange)
+            }
+
+            is Credit? -> listeners.sendEvent(PillarboxPlayer.EVENT_CREDIT_CHANGED) { listener ->
+                listener.onCreditChanged(timeRange)
+            }
+
+            is BlockedTimeRange -> listeners.sendEvent(PillarboxPlayer.EVENT_BLOCKED_TIME_RANGE_REACHED) { listener ->
+                listener.onBlockedTimeRangeReached(timeRange)
+                handleBlockedTimeRange(timeRange)
```
[the remark this answers is not in the document]
Comment (StaehliJ) [reply pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/PillarboxExoPlayer.kt:361]: Totally ;)

**Review by MGaetan89 — APPROVED** [review]

**Pull request merged**